app/connectors/propertyLinking/PropertyLinkConnector.scala

```diff
@@ -72,38 +72,12 @@ class PropertyLinkConnector @Inject()(config: ServicesConfig, http: WSHttp)(impl
           "pageSize" -> pagination.pageSize.toString,
           "requestTotalRowCount" -> pagination.requestTotalRowCount.toString)
     )
-  }
 
-  def getMyClientsPropertyLinks(
-                                 searchParams: GetPropertyLinksParameters,
-                                 pagination: PaginationParams,
-                                 representationStatusFilter: Seq[RepresentationStatus] = Seq(RepresentationApproved, RepresentationPending)
-                               )(implicit hc: HeaderCarrier): Future[OwnerAuthResult] = {
-    val ownerAuthResult = http.GET[OwnerAuthResult](
-      s"$baseUrl/agent/property-links",
-      List(
-        searchParams.address.map("address" -> _),
-        searchParams.baref.map("baref" -> _),
-        searchParams.agent.map("agent" -> _),
-        searchParams.status.map("status" -> _),
-        Some("sortField" -> searchParams.sortfield.toString),
-        Some("sortOrder" -> searchParams.sortorder.toString)
-      ).flatten ++
-        List(
-          "startPoint" -> pagination.startPoint.toString,
-          "pageSize" -> pagination.pageSize.toString,
-          "requestTotalRowCount" -> pagination.requestTotalRowCount.toString)
-    )
-
-    def validAgent(agent: OwnerAuthAgent): Boolean =
-      agent.status.fold(false) { status =>
-        representationStatusFilter.map(_.name.toUpperCase).contains(status.toUpperCase)
-      }
+    filterAgents(ownerAuthResult, representationStatusFilter)
```

Comment:
dont think this works the way you think it do?

Reply:
as discussed